src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerUdpBatcher.cs

```diff
@@ -56,106 +56,109 @@ public JaegerUdpBatcher(JaegerExporterOptions options)
+            }
+
+            var flushedSpanCount = 0;
+
+            await this.flushLock.WaitAsync().ConfigureAwait(false);
+            try
+            {
+                // flush if current batch size plus new span size equals or exceeds max batch size
+                if (this.batchByteSize + spanSize >= this.maxPacketSize)
+                {
+                    flushedSpanCount = await this.FlushAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    this.maxFlushIntervalTimer.Enabled = true;
+                }
+
+                // add span to batch and wait for more spans
+                this.currentBatch.Add(span);
+                this.batchByteSize += spanSize;
+            }
+            finally
+            {
+                this.flushLock.Release();
+            }
 
             return flushedSpanCount;
         }
 
         public async Task<int> FlushAsync(CancellationToken cancellationToken)
-        {
-            try
-            {
-                await this.flushLock.WaitAsync().ConfigureAwait(false);
-
-                this.maxFlushIntervalTimer.Enabled = false;
-
-                int n = this.currentBatch.Count;
-
-                if (n == 0)
-                {
-                    return 0;
-                }
-
-                try
-                {
-                    await this.SendAsync(this.Process, this.currentBatch, cancellationToken).ConfigureAwait(false);
-                }
-                finally
-                {
-                    this.currentBatch.Clear();
-                    this.batchByteSize = this.processByteSize.Value;
-                }
-
-                return n;
-            }
-            finally
-            {
-                this.flushLock.Release();
-            }
+        {
+            this.maxFlushIntervalTimer.Enabled = false;
+
+            int n = this.currentBatch.Count;
+
+            if (n == 0)
+            {
+                return 0;
+            }
+
+            try
+            {
+                await this.SendAsync(this.Process, this.currentBatch, cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                this.currentBatch.Clear();
+                this.batchByteSize = this.processByteSize.Value;
+            }
+
+            return n;
         }
 
-        public virtual Task<int> CloseAsync(CancellationToken cancellationToken) => this.FlushAsync(cancellationToken);
+        public virtual Task<int> CloseAsync(CancellationToken cancellationToken) => this.LockedFlushAsync(cancellationToken);
 
         public void Dispose()
         {
             // Do not change this code. Put cleanup code in Dispose(bool disposing).
             this.Dispose(true);
         }
 
+        protected async Task<int> LockedFlushAsync(CancellationToken cancellationToken)
```

Comment:
Agree with @MikeGoldsmith this probably isn't needed. But if it is kept, I think maybe better to make it private. Given `flushLock` is private, if someone was to override this in a derived type they wouldn't really be able to participate in the locking anyway.

Reply:
NM, I see it now. If you take the lock in Append and then call into Flush because the buffer is full Flush won't be able to take the lock again. IMO `LockedFlushAsync` is very confusing, maybe a flag passed in (ex: bool lockHeld) might be more concise?

Reply:
Yeah, I agree this should be private too

```suggestion
        private async Task<int> LockedFlushAsync(CancellationToken cancellationToken)
```

Reply:
Thanks Mike. I have committed these changes. I was trying to fix the whitespace consolidating in a single commit but it seems that the original file came with different line endings mixed in the file. Currently I have pushed changes with only LF in this file